### DISCUSSION TITLE
How to stop after reaching certain time step instead of stop_value

### DISCUSSION THREAD

**Discussion by Seraphli** [opendilab/DI-engine#607] [Q&A] [answered]
When tests are stopped at `stop_value`, the results can hardly be aggregated because of the different finished time steps. Is there a way to set the total time steps?

**Comment by PaParaZz1** [comment, marked as answer]
1. set `stop_value` to `math.inf`
2. set `max_env_step` or `max_train_iter` like this [implementation](https://github.com/opendilab/DI-engine/blob/main/ding/entry/serial_entry.py#L117)
3. For new task pipeline, you can use [termination_checker](https://github.com/opendilab/DI-engine/blob/main/ding/framework/middleware/functional/termination_checker.py) middleware

**Reply by Seraphli** [reply]
Thanks!
